modules/build/src/main/scala/scala/build/CrossSources.scala

```diff
@@ -126,6 +126,7 @@ object CrossSources {
     inputs: Inputs,
     preprocessors: Seq[Preprocessor],
     logger: Logger,
+    suppressDirectivesInMultipleFilesWarning: Option[Boolean],
```

Comment:
I wonder if it's necessary to pass the flag all the way here like this, since it's already in `buildOptions`... 
`buildOptions` are initialised later in this method (line 218 as of this comment), I think it should be retrievable from there...
the warning would have to be printed slightly later though. which I think should be fine.